tests/poll.rs

```diff
@@ -310,6 +311,67 @@ fn registry_operations_are_thread_safe() {
     handle3.join().unwrap();
 }
 
+#[test]
+fn poll_ok_after_cancelling_pending_ops() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+    let address = listener.local_addr().unwrap();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let registry1 = Arc::clone(&registry);
+
+    let barrier = Arc::new(Barrier::new(2));
+    let barrier1 = Arc::clone(&barrier);
+
+    registry
+        .register(&mut listener, ID1, Interest::READABLE)
+        .unwrap();
+
+    // Call a dummy poll just to submit an afd poll request
+    poll.poll(&mut events, Some(Duration::from_millis(0)))
+        .unwrap();
+
+    // this reregister will cancel the previous pending poll op
+    registry
+        .reregister(&mut listener, ID1, Interest::READABLE | Interest::WRITABLE)
```

Comment:
You might want to use `ID2`/`ID3` so you can ensure below that the proper event got returned in `expect_events` below.

Reply:
Good idea, will change it.